src/frontend/src/lib/utils/transaction.utils.ts

```diff
@@ -29,3 +34,34 @@ export const mapTransactionIcon = ({
 					? IconSend
 					: IconReceive;
 };
+
+/**
+ * Group a list of transactions by date.
+ *
+ * It assumes that the transactions are already sorted by timestamp in descending order.
+ * It does not sort the transactions, nor the groups.
+ * If a transaction has no timestamp, it will be grouped under the key 'undefined'.
+ *
+ * @param transactions - List of transactions to group.
+ * @returns Object where the keys are the date and the values are the transactions for that date.
+ */
+export const groupTransactionsByDate = <
```

Comment:
Can you type the return type?

Reply:
done! any better naming?

`export type AnyTransactionUi = BtcTransactionUi | EthTransactionUi | IcTransactionUi;`